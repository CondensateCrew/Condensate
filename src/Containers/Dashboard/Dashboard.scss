@import "../../css/variables";
@import "../../css/index.scss";

.dashboard {
  display: grid;
  grid-template-areas: 'aside header' 'aside main';
  grid-template-columns: 1.08fr 2.92fr;
  grid-template-rows: 1fr 8fr;
  aside {
    grid-area: aside;
    h1 {
      font-size: $big-text;
      text-align: center;
    }
  }
  header {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
  main {
    height: 100%; width: 100%;
    overflow-y: scroll;
  }
}

.dashboard-brainstorm-form-div {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $one-line-border-radius;
  padding: $one-line-padding-space;
  h3 {
    font-size: $small-text;
    font-weight: 700;
    text-align: left;
    width: 100%;
  }
  button {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: center;
    background: $primary-gradient;
    border-radius: $round-border-radius;
    padding: $extra-small-space;
    box-shadow: $external-shadow;
    margin-right: $small-space;
    img {
      width: $big-text;
      opacity: 0.4;
      &:hover {
        opacity: 1;
        transform: scale(1.1);
      }
      &:active {
        transform: scale(0.9);
      }
    }
  }
}

.aside {
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: justify;
}
