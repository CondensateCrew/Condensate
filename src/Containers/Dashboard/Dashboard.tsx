import React, { useState } from 'react';
import './Dashboard.scss';
import { Category } from 'interfaces';
import BrainstormContainer from '../BrainstormContainer/BrainstormContainer';
import SearchBar from '../SearchBar/SearchBar';
import CategoriesFilter from '../CategoriesFilter/CategoriesFilter';
import BrainstormForm from '../BrainstormForm/BrainstormForm';
import UserProfile from '../../Components/UserProfile/UserProfile';
import addIcon from '../../assets/add.svg';

const categories: Category[] = [
  {id: 1, name: 'Tech'},
  {id: 2, name: 'Education'},
  {id: 3, name: 'Environment'},
  {id: 4, name: 'Money'},
  {id: 5, name: 'Food'}];

const Dashboard: React.FC = () => {
  const [ brainstormFormState, setBrainstormForm ] = useState<boolean>(false)
  let brainstormForm;

  const toggleMenu = () => {
    setBrainstormForm(!brainstormFormState)
  }

  brainstormFormState ? brainstormForm = (<BrainstormForm brainstormFormState={brainstormFormState}
    cancel={setBrainstormForm}/>)
  : brainstormForm = (<div className='dashboard-brainstorm-form-div'>
      <img onClick={toggleMenu} className='add-icon-dashboard' alt='add-icon' src={addIcon}/>
      <h3>create new brainstorm</h3>
      </div>)


  return (
    <main className="dashboard">
      <aside className='aside'>
        <h1>condensate</h1>
        <UserProfile firstName={'Alan'} lastName={'Birds'}/>
        {brainstormForm}
      </aside>
      <header>
        <CategoriesFilter {...{categories}} />
        <SearchBar />
      </header>
      <BrainstormContainer />
    </main>
  )
}

export default Dashboard;
