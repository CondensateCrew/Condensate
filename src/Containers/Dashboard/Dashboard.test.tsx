import React from 'react';
import Dashboard from './Dashboard';
import { shallow } from 'enzyme';

jest.mock('react-redux', () => ({
  useSelector: () => ['Tech']
}));

describe('Dashboard component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Dashboard />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot when the new brainstorm button has been clicked', () => {
    wrapper = shallow(<Dashboard />);

    wrapper.find('.dashboard-brainstorm-form-div').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
