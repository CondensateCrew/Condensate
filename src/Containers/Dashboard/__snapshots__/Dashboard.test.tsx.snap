// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should match the snapshot 1`] = `
<main
  className="dashboard"
>
  <aside
    className="aside"
  >
    <h1>
      condensate
    </h1>
    <UserProfile />
    <div
      className="dashboard-brainstorm-form-div"
    >
      <button>
        <img
          alt="add-icon"
          onClick={[Function]}
          src="add.svg"
        />
      </button>
      <h3>
        start new brainstorm
      </h3>
    </div>
  </aside>
  <header>
    <CategoriesFilter
      categories={
        Array [
          "Tech",
        ]
      }
    />
    <SearchBar />
  </header>
  <main>
    <BrainstormContainer />
  </main>
</main>
`;

exports[`Dashboard component should match the snapshot when the new brainstorm button has been clicked 1`] = `
<main
  className="dashboard"
>
  <aside
    className="aside"
  >
    <h1>
      condensate
    </h1>
    <UserProfile />
    <div
      className="dashboard-brainstorm-form-div"
    >
      <button>
        <img
          alt="add-icon"
          onClick={[Function]}
          src="add.svg"
        />
      </button>
      <h3>
        start new brainstorm
      </h3>
    </div>
  </aside>
  <header>
    <CategoriesFilter
      categories={
        Array [
          "Tech",
        ]
      }
    />
    <SearchBar />
  </header>
  <main>
    <BrainstormContainer />
  </main>
</main>
`;
